### PR TITLE
ui: metrics time selector unit test coverage

### PR DESCRIPTION
**Pull request by vladlos**
converted `<TimeScaleDropdown />` to functional component due to problems with `withRouter` hoc in test env
added unit tests for `<TimeScaleDropdown />` and `<TimeFrameControls />`

Release note (ui): none

### CODE REVIEW

**Review by koorosh — CHANGES_REQUESTED** [review]
Got an error on project build:
```
ERROR: cockroachdb/cockroach/pkg/ui/src/views/cluster/components/controls/controls.spec.tsx:41:4 - file should end with a newline
```
Please take a look at this error.

File: pkg/ui/src/views/cluster/components/controls/controls.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// Copyright 2018 The Cockroach Authors.
```
Comment (koorosh) [inline pkg/ui/src/views/cluster/components/controls/controls.spec.tsx:1]: Please, update year. 2018 -> 2020

**Review by koorosh — COMMENTED** [review]
@elkmaster , I have a concern about `TimeFrameControls` component and props it exposes.
- `disabledArrows` prop makes you iterate three times over the same array to check if buttons have to be disabled.
-  Would it be more clear to name props as `onButtonClick` instead of `onArrowClick`
- it is only three buttons in the component and I suppose it can be more meaningful to change  `disabledArrows` prop to three props like: `isBackButtonDisabled`, `isForwardButtonDisabled`, and `isNowButtonDisabled` props.

File: pkg/ui/src/views/cluster/containers/timescale/index.tsx [hunk 2]
```diff
@@ -62,25 +69,93 @@ export const getTimeLabel = (currentWindow?: timewindow.TimeWindow, windowSize?:
...
+      setQueryParamsByDates(newSettings.windowSize, moment());
+      setTimeScale({ ...newSettings, key: newTimescaleKey.value });
     }
-  }
+  };
 
-  arrowClick = (direction: ArrowDirection) => {
-    const { currentWindow, currentScale } = this.props;
-    const windowSize: any = moment.duration(moment(currentWindow.end).diff(currentWindow.start));
+  const arrowClick = (direction: ArrowDirection) => {
```
[the remark this answers is not in the document]
Comment (elkmaster) [reply pkg/ui/src/views/cluster/containers/timescale/index.tsx:154]: Arrow direction is not used as dropdown icon direction. Not great naming due to the fact as on previous designs it was arrows. But I would do not refactor that part in terms of this pr as this is about tests only. Let's revisit it in terms of integrating DS component here.

File: pkg/ui/src/views/cluster/containers/timescale/timescale.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+import React from "react";
+// import { assert } from "chai";
```
Comment (koorosh) [inline pkg/ui/src/views/cluster/containers/timescale/timescale.spec.tsx:12]: Remove commented imports

File: pkg/ui/src/views/cluster/containers/timescale/timescale.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+  "#/metrics/overview/cluster/cluster?start=1584528492&end=1584529092", // 10 minutes
+  "#/metrics/overview/cluster?start=1583319565&end=1584529165", // 2 weeks
+  "#/metrics/overview/node/1", // Node 1 - Past 10 minutes
+  `#/metrics/overview/node/2?start=${moment()
+    .subtract(10, "minutes")
+    .format("X")}&end=${moment().format("X")}`, // Node 2 - Past 10 minutes
+  "#/metrics/overview/node/3?start=1584528726&end=1584529326", // Node 3 - 10 minutes
+];
+
+const dateFormat = "MMM DD,";
```
Comment (koorosh) [inline pkg/ui/src/views/cluster/containers/timescale/timescale.spec.tsx:50]: Would it be better to reuse `dateFormat` and `timeFormat` by exporting them and importing them in tests? Will reduce code duplication.

File: pkg/ui/src/views/cluster/containers/timescale/index.tsx [hunk 5]
```diff
@@ -62,25 +69,93 @@ export const getTimeLabel = (currentWindow?: timewindow.TimeWindow, windowSize?:
...
+) => {
+  const dateFormat = "MMM DD,";
+  const timeFormat = "h:mmA";
+  if (currentScale.key === "Custom" && currentWindow) {
+    const isSameStartDay = moment(currentWindow.start).isSame(moment(), "day");
+    const isSameEndDay = moment(currentWindow.end).isSame(moment(), "day");
+    return {
+      dateStart: isSameStartDay
+        ? ""
+        : moment.utc(currentWindow.start).format(dateFormat),
```
[the remark this answers is not in the document]
Comment (elkmaster) [reply pkg/ui/src/views/cluster/containers/timescale/index.tsx:84]: good catch, added utc for L79-80

File: pkg/ui/src/views/cluster/containers/timescale/timescale.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    assert.equal(
+      wrapper.props().currentScale,
+      timewindow.availableTimeScales["Past 10 Minutes"],
+    );
+  });
+
+  it("getTimeRangeTitle must return title Past 10 Minutes", () => {
+    const title = getTimeRangeTitle(state.currentWindow, state.currentScale);
+    const wrapper = makeTimeScaleDropdown(state);
+    console.log(wrapper.find(".trigger .Select-value-label"));
```
Comment (koorosh) [inline pkg/ui/src/views/cluster/containers/timescale/timescale.spec.tsx:101]: Remove `console.log`

**Comment by elkmaster** [comment]
> @elkmaster , I have a concern about `TimeFrameControls` component and props it exposes.
> 
> * `disabledArrows` prop makes you iterate three times over the same array to check if buttons have to be disabled.
> * Would it be more clear to name props as `onButtonClick` instead of `onArrowClick`
> * it is only three buttons in the component and I suppose it can be more meaningful to change  `disabledArrows` prop to three props like: `isBackButtonDisabled`, `isForwardButtonDisabled`, and `isNowButtonDisabled` props.

I agree but would like to revisit this in terms of other ticket. This pr is about test coverage only.

File: pkg/ui/src/views/cluster/containers/timescale/timescale.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+} from "./index";
+import * as timewindow from "src/redux/timewindow";
+// import sinon from "sinon";
+import moment from "moment";
+import { refreshNodes } from "src/redux/apiReducers";
+import "src/enzymeInit";
+import { MemoryRouter } from "react-router";
+import TimeFrameControls from "../../components/controls";
+import RangeSelect from "../../components/range";
+import { expect, assert } from "chai";
```
Comment (koorosh) [inline pkg/ui/src/views/cluster/containers/timescale/timescale.spec.tsx:29]: What is a reason to use `expect` for assertions? By default `assert` is used across the entire project.

**Review by koorosh — CHANGES_REQUESTED** [review]
Tests look good to me, but code still needs to be cleaned up. Please take a look at my comments.

File: pkg/ui/src/views/cluster/components/controls/controls.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+import TimeFrameControls from "../../components/controls";
+import { RangeSelectProps } from "./index";
+
+describe("<TimeFrameControls>", function() {
+  let spy: sinon.SinonSpy;
+
+  const makeTimeScaleDropdown = (props: RangeSelectProps) => shallow((<TimeFrameControls {...props} />));
+
+  beforeEach(function() {
+    spy = sinon.spy();
```
[the remark this answers is not in the document]
Comment (elkmaster) [reply pkg/ui/src/views/cluster/components/controls/controls.spec.tsx]: that was original intent, but it is not worked well, so I switched to class based tests. Removed leftovers

**Review by koorosh — APPROVED** [review]
LGTM

**Review by dhartunian — APPROVED** [review]

**Pull request merged**